### PR TITLE
Update CXXFLAGS

### PR DESCRIPTION
**Pull request by Scrumplex**
By switching to -O2 we might be able to avoid random issues like those we worked around in #616

### CODE REVIEW

**Review by timoreo22 — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,21 +34,24 @@ set(CMAKE_C_STANDARD_REQUIRED true)
...
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Werror=return-type")
 
 # Fix build with Qt 5.13
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DQT_NO_DEPRECATED_WARNINGS=Y")
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DQT_DISABLE_DEPRECATED_BEFORE=0x050C00")
 
+# set CXXFLAGS for build targets
+set(CMAKE_CXX_FLAGS_DEBUG "-O1 ${CMAKE_CXX_FLAGS}")
+set(CMAKE_CXX_FLAGS_RELEASE "-O2 -D_FORTIFY_SOURCE=2 ${CMAKE_CXX_FLAGS}")
```
Comment (Scrumplex) [inline CMakeLists.txt]: not sure if we want [`-D_FORTIFY_SOURCE=2`](https://stackoverflow.com/questions/13517526/difference-between-gcc-d-fortify-source-1-and-d-fortify-source-2) for both debug and release builds. We had no issues so far.

**Review by d-513 — APPROVED** [review]

**Comment by Scrumplex** [comment]
I don't want to rush this for 1.3.0. Let's take our time.

**Comment by Scrumplex** [comment]
Maybe `-fpie` as well?

File: CMakeLists.txt (hunk 1 above)
Comment (txtsd) [reply CMakeLists.txt]: Well we don't really have very good regression tests, but like you said we haven't had any related issue so far.

**Comment by Scrumplex** [comment]
Okay these changes aren't very radical. I think we can squeeze them into 1.3.0

**Review by flowln — APPROVED** [review]
`-fuse-ld=mold` wen? :P

**Pull request merged**